src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/PopupControlService.cs

```diff
@@ -1475,6 +1475,44 @@ private void AddPoints(PointList points, in NativeMethods.RECT rect, bool rectIs
                 }
             }
 
+            // Test whether the current mouse point lies within the convex hull
+            internal bool ContainsMousePoint()
+            {
+                IInputElement rootElement = _source.RootVisual as IInputElement;
+                if (rootElement != null)
+                {
+                    // get the coordinates of the current mouse point, relative to our PresentationSource
+                    System.Windows.Point pt = Mouse.PrimaryDevice.GetPosition(rootElement);
+
+                    // check whether the point lies within the hull
+                    return ContainsPoint(_source, (int)pt.X, (int)pt.Y);
```

Comment:
Could there be a loss of precision here? How about changing the ContainsPoint method argument from int to double?

Reply:
ContainsPoint requires 16-bit integers, which is what GetPosition supplies (albeit wrapped in a Point whose fields have type double).  See the [comment](https://github.com/dotnet/wpf/pull/5931/files/e43234e81b48430b36ff24d0d29a22fe2f9b05dd#diff-e4eeb1239fe1089ce77795654002e57e7c12c2b616962d9369b70878e74b8e95R1233) for class ConvexHull.